### ISSUE TITLE
Code quality improvement tips

### ISSUE THREAD

**Issue by MathieuAA** [justrelease/justrelease#3]
Sorry if I'm a bit harsh, but I'm a bit tired.

I'm not going to point everything that bothers me, just the basic stuff.

Don't hesitate abusing private fields =) Like [here ](https://github.com/justrelease/justrelease/blob/master/src/main/java/com/justrelease/JustRelease.java#L34) for instance.

The code needs some cleaning-up [here](https://github.com/justrelease/justrelease/blob/master/src/main/java/com/justrelease/JustRelease.java#L34), you can either replace it with a `else`, or exit the method (much better).

Your code needs some more cleaning as the `#release` method is **way** too complex (too many lines, refactorable code, etc.).

In [this](https://github.com/justrelease/justrelease/blob/master/src/main/java/com/justrelease/JustReleaseCLI.java) file, you should not try-catch, and display the help (and then exit with the system helper), just throw it with a nice message (if the user wants help, I guess he/she can type one more line in the shell).

The code needs refactoring, just search for the 'cmd' in the same file (a bad word, you should use 'command') word, and try to make the occurrences nearer to each other.

In [this](https://github.com/justrelease/justrelease/blob/master/src/main/java/com/justrelease/project/type/AbstractProjectInfo.java) file, `releaseConfig` needs to be hidden!

In the MavenProject class, you use a setter on a field belonging to an abstract class extended by MavenProject. Even if it has no visible incidence, it is considered bad practice...

To sum up (I didn't go through every file):
- Don't use the default visibility in concrete ( != interface) classes;
- Use consistent styling (the `{}` either need to be put after every if/for/etc instructions everywhere, or no where, just be consistent);
- Use private methods (or delegate) to make your methods lighter.

I suggest you use Sonar as it can help you with code quality.

By the way, your project looks promising, and I'll test it when I can!

**Comment by bilalyasar** [comment]
Thanks for the great feedback. We have been doing refactoring already. Can you please try releasing your software with justrelease?

**Comment by MathieuAA** [comment]
Yes, I'll definitely will for the version 1.4.3!

**Comment by mesutcelik** [comment]
many thanks. waiting for your feedback.

**Comment by MathieuAA** [comment]
Maybe not for v1.4.3, but I'll try to test it add make a release for a dummy project. Good work guys!
